draft-bbs-signatures.md

```diff
@@ -818,27 +831,156 @@ This operation describes how to encode a signature to an octet string.
-- s (REQUIRED), an integer representing a valid scalar value within the range of 0 < e < r.
+- signature (REQUIRED), a valid signature, in the form (A, e, s), where
+                        A a point in G1 and e, s non-zero scalars mod r.
 
 Outputs:
 
 - signature_octets, octet string.
 
 Procedure:
 
-1. A_octets = point_to_octets_g1(A)
+1. (A, e, s) = signature
+
+2. A_octets = point_to_octets_g1(A)
+
+3. e_octets = I2OSP(e, octet_scalar_length)
+
+4. s_octets = I2OSP(s, octet_scalar_length)
+
+5. return (a_octets || e_octets || s_octets)
+```
+
+### OctetsToProof
+
+This operation describes how to decode an octet string representing a proof, validate it and return the underlying components that make up the proof value.
+
+The proof value outputted by this operation consists of the following components, in that order:
+
+1. Three (3) valid points of the G1 subgroup, each of which must not equal the identity point.
+2. Five (5) integers representing scalars in the range of 1 to r-1 inclusive.
+3. A set of integers representing scalars in the range of 1 to r-1 inclusive, corresponding to the un-revealed from the proof message commitments. This set can be empty (i.e., "()").
+
+```
+proof = octets_to_proof(proof_octets)
+
+Inputs:
+
+- proof_octets (REQUIRED), octet string of the form outputted from the
+                           proof_to_octets operation.
+
+Parameters:
+
+- r (REQUIRED), non-negative integer. The prime order of the G1 and
+                G2 groups, defined by the ciphersuite.
+- octet_scalar_length (REQUIRED), non-negative integer. The length of
+                                  a scalar octet representation, defined
+                                  by the ciphersuite.
+- octet_point_length (REQUIRED), non-negative integer. The length of
+                                 a point in G1 octet representation,
+                                 defined by the ciphersuite.
+
+Outputs:
+
+- proof, a proof value in the form described above or INVALID
+
+Procedure:
+
+1. proof_len_floor = 3 * octet_point_length + 5 * octet_scalar_length
+
+2. if length(proof_octets) < proof_len_floor, return INVALID
+
+// Points (i.e., (A', Abar, D) in ProofGen) de-serialization.
+3. index = 0
+
+4. for i in range(0, 2):
+
+5.     end_index = index + octet_point_length - 1
+
+6.     A_i = octets_to_point_g1(proof_octets[index..end_index])
+
+7.     if A_i is INVALID or Identity_G1, return INVALID
+
+8.     index += octet_point_length
+
+// Scalars (i.e., (c, e^, r2^, r3^, s^, (m^_j1, ..., m^_jU)) in
+// ProofGen) de-serialization.
+9. j = 0
+
+10. while index < length(proof_octets):
+
+11.     end_index = index + octet_scalar_length - 1
+
+12.     s_j = OS2IP(proof_octets[index..end_index])
+
+13.     if s_j = 0 or if s_j >= r, return INVALID
+
+14.     index += octet_scalar_length
+
+15.     j += 1
+
+16. if index is not equal to length(proof_octets), return INVALID
+
+17. Let msg_commitments be an empty set (i.e., msg_commitments = ())
```

Comment:
Otherwise from a data representation perspective serialized as JSON you could take the current syntax to mean

```
proof = {
   A_0: ....,
   .....
   msg_commitments: {
      s_5,
      ...
   }
}
```

Instead

```
proof = {
   A_0: ....,
   .....
   s_5,
   ...
}
```

Reply:
I think that in the rest of the spec we use it as the first JSON example you listed. For example, in [proofGen](https://identity.foundation/bbs-signature/draft-bbs-signatures.html#name-proofgen), in step 25 we compute,
```
25. proof = (A', Abar, D, c, e^, r2^, r3^, s^, (m^_j1, ..., m^_jU))
```
And in [ProofVerify](https://identity.foundation/bbs-signature/draft-bbs-signatures.html#name-proofverify), at step 4, we (currently) parse the proof as,
```
4. (A', Abar, D, c, e^, r2^, r3^, s^, (m^_j1,...,m^_jU)) = proof
```
Show the options I see is
1. We update this on those places as well (will be really easy to do in this PR)
2. We keep using the same notation

I'm fine with both. I don't think it make much difference from an implementations perspective. From an aesthetics perspective i may go with option 1, so there is some distinction between the different parts of the proof, but really not filling strong about it.

Reply:
Im actually fine with either so long as we are consistent, suffice to say if you think we are already being consistent w.r.t  option 1 then we can leave it as is.

Reply:
There is a pending suggested change attached to this comment also which isn't entirely related to this conversation, once we have a resolution on that I'm happy to approve

Reply:
Yeap! Updated based on suggestion. Thanks!